src/megaapi_impl.cpp

```diff
@@ -8236,6 +8244,103 @@ void MegaApiImpl::getversion_result(int versionCode, const char *versionString,
     fireOnRequestFinish(request, MegaError(e));
 }
 
+void MegaApiImpl::getlocalsslcertificate_result(m_time_t ts, string *certdata, error e)
+{
+    MegaError megaError(e);
+    if(requestMap.find(client->restag) == requestMap.end()) return;
+    MegaRequestPrivate* request = requestMap.at(client->restag);
+    if(!request || (request->getType() != MegaRequest::TYPE_GET_LOCAL_SSL_CERT)) return;
+
+    if (!e)
+    {
+        MegaStringMapPrivate *datamap = new MegaStringMapPrivate();
+
+        string result;
+        const char *data = certdata->data();
+        const char *end = strstr(data, ";");
+        if (!end)
+        {
+            delete datamap;
+            fireOnRequestFinish(request, MegaError(API_EINTERNAL));
+            return;
+        }
+
+        result.append("-----BEGIN PRIVATE KEY-----\n");
+        while (data < end)
+        {
+            int remaining = end - data;
+            if (remaining > 64)
+            {
+                result.append(data, 64);
+                data += 64;
+            }
+            else
+            {
+                result.append(data, remaining);
+                data += remaining;
+            }
+            result.append("\n");
+        }
+        result.append("-----END PRIVATE KEY-----\n");
+        datamap->set("key", result.c_str());
+
+        int i = 0;
+        while (*data++)
+        {
+            end = strstr(data, ";");
+            if (!end)
+            {
+                end = data + strlen(data);
+                if (data == end)
+                {
+                    delete datamap;
+                    fireOnRequestFinish(request, MegaError(API_EINTERNAL));
+                    return;
+                }
+            }
+            result = "-----BEGIN CERTIFICATE-----\n";
+            while (data < end)
```

Comment:
Maybe this block to restore the format of the key/certificate could be done in an auxiliary function, so we don't duplicate the code for the private key, the main certificate and the intermediate ones. Not a big deal indeed, since the headers need to be different.